pkg/installmanager/installmanager.go

```diff
@@ -1810,3 +1820,62 @@ func (m *InstallManager) overrideCredentialsModeOnInstallConfig(credentialsMode
 
 	return nil
 }
+
+func (m *InstallManager) patchAzureOverrideCreds() error {
+	overrideCredsPath := filepath.Join(m.WorkDir, "manifests", overrideCredsYAML)
+	overrideCredsBytes, err := ioutil.ReadFile(overrideCredsPath)
+	if err != nil {
+		return errors.Wrap(err, "error reading 99_cloud-creds-secret.yaml")
+	}
+	c, err := yamlutils.Decode(overrideCredsBytes)
+	if err != nil {
+		return errors.Wrap(err, "unable to decode 99_cloud-creds-secret.yaml content")
+	}
+	if isAzureCreds, _ := yamlutils.Test(c, "/metadata/name", "azure-credentials"); !isAzureCreds {
+		return errors.Wrap(err, "wrong input, secret name is not azure-credentials in 99_cloud-creds-secret.yaml")
+	}
+	region, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "region", filepath.Join(m.WorkDir, "manifests", clusterConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep region from manifests")
+	}
+	infraName, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "infrastructureName", filepath.Join(m.WorkDir, "manifests", clusterInfraConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep infrastructureName from manifests")
+	}
+	resourceGroupName, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "resourceGroupName", filepath.Join(m.WorkDir, "manifests", clusterInfraConfigYAML))).Output()
```

Comment:
Rather than spawning subshells, let's get these values "natively".

yamlpatch isn't the easiest to use, but here's an example of retrieving a nested value: https://go.dev/play/p/qV4eam0zsFY

Reply:
Thanks for the tips! I’ll remove subshell.

Reply:
I included package github.com/tidwall/gjson since I found it is really simple to use, pls. let me know if it is NOT ok.

Reply:
I'm okay with it. We have some part of tidwall's suite in our go.sum already for some reason, so :shrug: 

@suhanime @abutcher @dlom do y'all see any concerns pulling this lib in?

Reply:
No concerns, looks to be a quite popular lib.